hatch/src/generators/tup/builder.rs

```diff
@@ -0,0 +1,138 @@
+use assets::{PlatformKind, ProjectAsset, TupKind};
+use constants::CATCH_HEADER_URL;
+use generators::tup::catch_header::CatchHeader;
+use generators::tup::tupfile_ini::TupfileIni;
+use hatch_error::{HatchResult, NullError, ResultExt};
+use project::Project;
+use reqwest;
+use std::path::PathBuf;
+use generators::tup::tuprules::make_tuprules_string;
+use generators::tup::test_tupfile::make_test_tupfile_string;
+use generators::tup::tup_config::make_tup_config_string;
+use generators::tup::catch_definition::make_catch_definition_string;
+use generators::tup::tupfile::make_tupfile_string;
+use generators::tup::platform::{mac_os, linux, windows};
+
+pub struct Builder<'builder> {
+    project_path: PathBuf,
+    project: &'builder Project,
+    pub assets: Vec<ProjectAsset>,
+}
+
+impl<'builder> Builder<'builder> {
+    pub fn new(project_path: PathBuf, project: &'builder Project) -> Builder<'builder> {
+        Builder {
+            project_path,
+            project,
+            assets: Vec::new(),
+        }
+    }
+
+    pub fn collect_assets(builder: Builder) -> Vec<ProjectAsset> {
```

Comment:
besides these file being move to the tup generator module, I renamed some methods to be more buiderly, like `add_tup_config` instead of `tup_config`.